crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -2,29 +2,36 @@ use blockifier::abi::abi_utils::get_storage_var_address;
 use frame_support::{assert_err, assert_ok};
 use mp_felt::Felt252Wrapper;
 use mp_transactions::compute_hash::ComputeTransactionHash;
-use mp_transactions::{InvokeTransaction, InvokeTransactionV1};
+use mp_transactions::{DeclareTransactionV2, InvokeTransaction, InvokeTransactionV1};
 use pretty_assertions::assert_eq;
 use sp_runtime::traits::ValidateUnsigned;
 use sp_runtime::transaction_validity::{
     InvalidTransaction, TransactionSource, TransactionValidityError, ValidTransaction,
 };
-use starknet_api::api_core::{ContractAddress, Nonce, PatriciaKey};
+use starknet_api::api_core::{ClassHash, ContractAddress, Nonce, PatriciaKey};
 use starknet_api::hash::StarkFelt;
 use starknet_api::state::StorageKey;
 use starknet_api::transaction::{Event as StarknetEvent, EventContent, EventData, EventKey, TransactionHash};
-use starknet_core::utils::get_selector_from_name;
+use starknet_core::utils::{get_selector_from_name, UdcUniqueSettings, UdcUniqueness};
 use starknet_crypto::FieldElement;
+use test_log;
 
-use super::constants::{BLOCKIFIER_ACCOUNT_ADDRESS, MULTIPLE_EVENT_EMITTING_CONTRACT_ADDRESS, TEST_CONTRACT_ADDRESS};
+use super::constants::{
+    BLOCKIFIER_ACCOUNT_ADDRESS, MULTIPLE_EVENT_EMITTING_CONTRACT_ADDRESS, TEST_CONTRACT_ADDRESS, UDC_ADDRESS,
+    UDC_SELECTOR,
+};
 use super::mock::default_mock::*;
 use super::mock::*;
-use super::utils::sign_message_hash;
+use super::utils::{get_contract_class, sign_message_hash};
+use crate::tests::utils::get_udc_deployed_contract_address;
 use crate::tests::{
     get_invoke_argent_dummy, get_invoke_braavos_dummy, get_invoke_dummy, get_invoke_emit_event_dummy,
     get_invoke_nonce_dummy, get_invoke_openzeppelin_dummy, get_storage_read_write_dummy, set_nonce,
 };
 use crate::{Call, Config, Error, StorageView};
 
+const INCREASE_BALANCE_FUNCTION_SELECTOR: &str = "0x362398bec32bc0ebb411203221a35a0301193a96f317ebe5e40be9f60d15320";
```

Comment:
it's specific to a test case so we can move this constant there itself.

Reply:
done